### PR TITLE
Added support for programs APIs

### PR DESCRIPTION
**Pull request by steflindar**

### CODE REVIEW

File: src/main/java/com/lindar/dotmailer/vo/api/NewProgram.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package com.lindar.dotmailer.vo.api;
+
+import lombok.Data;
+
+import java.util.List;
+
+@Data
+public class NewProgram {
```
Comment (steven509) [inline src/main/java/com/lindar/dotmailer/vo/api/NewProgram.java]: This name is a bit confusing as it's not a new program, it's enrolment into a program?

**Review by steven509 — CHANGES_REQUESTED** [review]

File: src/main/java/com/lindar/dotmailer/api/ProgramResource.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    public Result<ProgramEnrolment> getEnrolment(String id) {
+        String path = pathWithParam(DefaultEndpoints.PROGRAM_ENROLMENT_BY_ID.getPath(), id);
+        return sendAndGet(path, ProgramEnrolment.class);
+    }
+
+    public Result<List<ProgramEnrolment>> getEnrolmentsByStatus(String status) {
+        String path = pathWithParam(DefaultEndpoints.PROGRAM_ENROLMENTS_BY_STATUS.getPath(), status);
+        return sendAndGetFullList(path, new TypeToken<List<ProgramEnrolment>>() {
+        });
+    }
```
Comment (steven509) [inline src/main/java/com/lindar/dotmailer/api/ProgramResource.java:45]: Perhaps we can use a enum for the status here? The Dotdigital API is a bit weird using the same endpoint for both and I can see that causing issues, eg, if you use a id here it'll have deserialisation issues. At least by having a enum it's a bit clearer what the acceptable values are

**Review by steven509 — CHANGES_REQUESTED** [review]

File: src/main/java/com/lindar/dotmailer/vo/api/NewProgram.java (hunk 1 above)
Comment (steflindar) [reply src/main/java/com/lindar/dotmailer/vo/api/NewProgram.java]: Ok, I can use NewProgramEnrolment if you agree

Comment (steven509) [reply src/main/java/com/lindar/dotmailer/vo/api/NewProgram.java]: Sounds good to me

**Review by steven509 — APPROVED** [review]

**Pull request merged**